### DISCUSSION TITLE
Netbox with/ Apache2 fail to start

### DISCUSSION THREAD

**Discussion by jl-duarte** [netbox-community/netbox#9185] [Q&A]
Hello.
I just installed Netbox with apache2 and I have 2 errors that I can't fix.
I have all the services running, and when I try to access by https to the ip:port I get an error "ERR_SSL_PROTOCOL_ERROR".
If I access by hhtp I see the default apache page "It works".

Any suggestions?
my apache2ctl -S 
root@netbox-srv:/opt/netbox-3.2.1/netbox/static# apache2ctl -S
VirtualHost configuration:
*:443                  is a NameVirtualHost
         default server www.sti.com (/etc/apache2/sites-enabled/default-ssl.conf:2)
         port 443 namevhost www.sti.com (/etc/apache2/sites-enabled/default-ssl.conf:2)
         port 443 namevhost 192.168.1.167 (/etc/apache2/sites-enabled/netbox.conf:1)
ServerRoot: "/etc/apache2"
Main DocumentRoot: "/var/www/html"
Main ErrorLog: "/var/log/apache2/error.log"
Mutex ssl-stapling: using_defaults
Mutex proxy: using_defaults
Mutex ssl-cache: using_defaults
Mutex default: dir="/var/run/apache2/" mechanism=default
Mutex watchdog-callback: using_defaults
Mutex ssl-stapling-refresh: using_defaults
PidFile: "/var/run/apache2/apache2.pid"
Define: DUMP_VHOSTS
Define: DUMP_RUN_CFG
User: name="www-data" id=33
Group: name="www-data" id=33
 
netbox.conf
<VirtualHost *:443>
    ProxyPreserveHost On
    # CHANGE THIS TO YOUR SERVER'S NAME
    ServerName 192.168.1.167
    SSLEngine on
    SSLCertificateFile /etc/ssl/certs/netbox.crt
    SSLCertificateKeyFile /etc/ssl/private/netbox.key
    Alias /static /opt/netbox-3.2.1/netbox/static

    <Directory /opt/netbox-3.2.1/netbox/static>
        Options Indexes FollowSymLinks MultiViews
        AllowOverride None
        Require all granted
    </Directory>
    <Location /static>
        ProxyPass !
    </Location>

    RequestHeader set "X-Forwarded-Proto" expr=%{REQUEST_SCHEME}
    ProxyPass / http://127.0.0.1:8001/
    ProxyPassReverse / http://127.0.0.1:8001/
</VirtualHost>

netbox service status
root@netbox-srv:/etc/apache2/sites-enabled# systemctl status netbox.service
● netbox.service - NetBox WSGI Service
     Loaded: loaded (/etc/systemd/system/netbox.service; enabled; vendor preset: enabled)
     Active: active (running) since Fri 2022-04-22 13:10:11 UTC; 43min ago
       Docs: https://netbox.readthedocs.io/en/stable/
   Main PID: 626 (gunicorn)
      Tasks: 9 (limit: 9290)
     Memory: 467.9M
     CGroup: /system.slice/netbox.service
             ├─626 /opt/netbox-3.2.1/venv/bin/python3.8 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gu>
             ├─811 /opt/netbox-3.2.1/venv/bin/python3.8 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gu>
             ├─812 /opt/netbox-3.2.1/venv/bin/python3.8 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gu>
             ├─813 /opt/netbox-3.2.1/venv/bin/python3.8 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gu>
             ├─814 /opt/netbox-3.2.1/venv/bin/python3.8 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gu>
             └─815 /opt/netbox-3.2.1/venv/bin/python3.8 /opt/netbox/venv/bin/gunicorn --pid /var/tmp/netbox.pid --pythonpath /opt/netbox/netbox --config /opt/netbox/gu>

abr 22 13:10:11 netbox-srv systemd[1]: Started NetBox WSGI Service.
abr 22 13:10:35 netbox-srv gunicorn[626]: [2022-04-22 13:10:35 +0000] [626] [INFO] Starting gunicorn 20.1.0
abr 22 13:10:36 netbox-srv gunicorn[626]: [2022-04-22 13:10:35 +0000] [626] [INFO] Listening at: http://127.0.0.1:8001 (626)
abr 22 13:10:36 netbox-srv gunicorn[626]: [2022-04-22 13:10:35 +0000] [626] [INFO] Using worker: gthread
abr 22 13:10:36 netbox-srv gunicorn[811]: [2022-04-22 13:10:35 +0000] [811] [INFO] Booting worker with pid: 811
abr 22 13:10:36 netbox-srv gunicorn[812]: [2022-04-22 13:10:35 +0000] [812] [INFO] Booting worker with pid: 812
abr 22 13:10:37 netbox-srv gunicorn[813]: [2022-04-22 13:10:35 +0000] [813] [INFO] Booting worker with pid: 813
abr 22 13:10:37 netbox-srv gunicorn[814]: [2022-04-22 13:10:35 +0000] [814] [INFO] Booting worker with pid: 814
abr 22 13:10:37 netbox-srv gunicorn[815]: [2022-04-22 13:10:36 +0000] [815] [INFO] Booting worker with pid: 815

journalctl -xe
root@netbox-srv:/etc/apache2/sites-enabled# journalctl -xe
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- The unit netbox-rq.service has entered the 'failed' state with result 'exit-code'.
abr 22 13:53:54 netbox-srv systemd[1]: netbox-rq.service: Scheduled restart job, restart counter is at 73.
-- Subject: Automatic restarting of a unit has been scheduled
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- Automatic restarting of the unit netbox-rq.service has been scheduled, as the result for
-- the configured Restart= setting for the unit.
abr 22 13:53:54 netbox-srv systemd[1]: Stopped NetBox Request Queue Worker.
-- Subject: A stop job for unit netbox-rq.service has finished
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- A stop job for unit netbox-rq.service has finished.
--
-- The job identifier is 6932 and the job result is done.
abr 22 13:53:54 netbox-srv systemd[1]: Started NetBox Request Queue Worker.
-- Subject: A start job for unit netbox-rq.service has finished successfully
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- A start job for unit netbox-rq.service has finished successfully.
--
-- The job identifier is 6932.
abr 22 13:53:58 netbox-srv python3[3314]: Error 111 connecting to localhost:6379. Connection refused.
abr 22 13:53:59 netbox-srv systemd[1]: netbox-rq.service: Main process exited, code=exited, status=1/FAILURE
-- Subject: Unit process exited
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- An ExecStart= process belonging to unit netbox-rq.service has exited.
--
-- The process' exit code is 'exited' and its exit status is 1.
abr 22 13:53:59 netbox-srv systemd[1]: netbox-rq.service: Failed with result 'exit-code'.
-- Subject: Unit failed
-- Defined-By: systemd
-- Support: http://www.ubuntu.com/support
--
-- The unit netbox-rq.service has entered the 'failed' state with result 'exit-code'.

I see netbox Failed, but i dont know why.

**Comment by candlerb** [comment]
Please can you edit your post, and put three backticks (` ``` `) on a line of their own, immediately before and after the config/console data you're quoting.  This will make them readable.

But even now, I can see you have two separate problems:
1. Netbox and netbox-rq are failing to start.  You need to find out why.  `journalctl -eu netbox` is your starting point.
2. Some problem with SSL in Apache, which is hard to diagnose with a mangled configuration file

It's also helpful if you can specify which OS platform you're running on - e.g. Ubuntu, CentOS etc - and version.
